impeller/core/texture_descriptor.h

```diff
@@ -66,6 +66,28 @@ struct TextureDescriptor {
     return IsMultisampleCapable(type) ? count > 1 : count == 1;
   }
 
+  constexpr bool operator==(const TextureDescriptor& other) const {
+    return size == other.size &&                          //
+           storage_mode == other.storage_mode &&          //
+           format == other.format &&                      //
+           usage == other.usage &&                        //
+           sample_count == other.sample_count &&          //
+           type == other.type &&                          //
+           compression_type == other.compression_type &&  //
+           mip_count == other.mip_count;
+  }
+
+  constexpr bool operator!=(const TextureDescriptor& other) const {
+    return size != other.size ||                          //
+           storage_mode != other.storage_mode ||          //
+           format != other.format ||                      //
+           usage != other.usage ||                        //
+           sample_count != other.sample_count ||          //
+           type != other.type ||                          //
+           compression_type != other.compression_type ||  //
+           mip_count != other.mip_count;
```

Comment:
Could this just be `return !(*this == other)`?

Reply:
I guess this is a struct ... does that work? I don't know C++ 😆

Reply:
Yeah it should work; classes are just structs with private members by default. :p

Reply:
Oh, duh. I thought you were saying to do a ptr comparison or something like an implict equal operator. I didn't realize you were specifically talking about using the `==` operation to define the `!=` operation. Makes sense!